src/Core/MySQL/MySQLJsonConv.cpp

```
@@ -0,0 +1,940 @@
+                {
+                    text.resize(MAX_DATETIME_LENGTH);
+                    (void) sprintf(text.data(), "%04d-%02d-%02d %02d:%02d:%02d.%06d",
+                        static_cast<Int32> (ym / 13),
+                        static_cast<Int32> (ym % 13),
+                        static_cast<Int32> (ymd % (1 << 5)),
+                        static_cast<Int32> (hms >> 12),
+                        static_cast<Int32> ((hms >> 6) % (1 << 6)),
+                        static_cast<Int32> (hms % (1 << 6)),
+                        frac);
+
+                }
+            }
+            out += ('"' + text +'"');
+            break;
+        }
+        default:
+        {
+            text = String(data, data_length);
+            out += '"';
+            escape(text, out);
+            out += '"';
+            break;
+        }
+    }
+}
+
+void JsonValue::toJsonString(String & out)
+{
+    switch (json_type)
+    {
+        case JsonEnumType::OBJECT:
+            out += '{';
+            for (size_t i = 0; i < element_count; i++)
+            {
+                if (i > 0)
+                {
+                    out += ", ";
+                }
+                out += '"';
+                String key = getObjectKey(i);
+                escape(key, out);
+                out += "\": ";
+                getJsonValueForElement(i).toJsonString(out);
+            }
+
+            out += '}';
+            break;
+
+        case JsonEnumType::ARRAY:
+            out += '[';
+            for (size_t i = 0; i < element_count; i++)
+            {
+                if (i > 0)
+                {
+                    out += ", ";
+                }
+                getJsonValueForElement(i).toJsonString(out);
+            }
+            out += ']';
+            break;
+
+        case JsonEnumType::DOUBLE:
+        {
+            // ostringstream will remove the tail '0' automatically
+            std::ostringstream oss; // STYLE_CHECK_ALLOW_STD_STRING_STREAM
```

Can we use WriteBuffer instead?

It makes sense, because stringstream:
- depends on locale;
- silently ignores errors and produces data loss on memory allocation failure;
- is slow;